### ISSUE TITLE
Broken as of 3pm Eastern today

### ISSUE THREAD

**Issue by jantman** [jantman/xfinity-usage#8]
Sometime between 3:00 and 3:30 PM Eastern (GMT-5) today this broke for me. It appears that they've redesigned the page and removed the amount used; it now only shows the amount remaining as text:

![usage](https://user-images.githubusercontent.com/273454/33583054-638dc45c-d926-11e7-91a5-f48b763cb5d2.png)

The resulting error is:

```
2017-12-04 15:30:01,727 [DEBUG xfinity_usage.py:161 - root.__init__() ] Getting browser instance...
2017-12-04 15:30:01,727 [DEBUG xfinity_usage.py:167 - root.run() ] Getting page...
2017-12-04 15:30:01,727 [DEBUG xfinity_usage.py:380 - root.get_browser() ] getting Chrome browser (local) with --headless
2017-12-04 15:30:03,060 [DEBUG xfinity_usage.py:404 - root.get_browser() ] returning browser
2017-12-04 15:30:03,060 [INFO xfinity_usage.py:351 - root.get() ] GET https://customer.xfinity.com/#/devices
2017-12-04 15:30:07,986 [INFO xfinity_usage.py:247 - root.get_usage_page() ] Not logged in; logging in now
2017-12-04 15:30:07,986 [INFO xfinity_usage.py:257 - root.get_usage_page() ] Trying to login...
2017-12-04 15:30:07,990 [INFO xfinity_usage.py:179 - root.do_login() ] Logging in )
2017-12-04 15:30:08,201 [DEBUG xfinity_usage.py:201 - root.do_login() ] Clicking "Remember Me"
2017-12-04 15:30:08,686 [ERROR xfinity_usage.py:339 - root.error_screenshot() ] Screenshot saved to: /home/jantman/scratch/webdriver_fail.png
2017-12-04 15:30:08,690 [ERROR xfinity_usage.py:340 - root.error_screenshot() ] Page title: Sign in to XFINITY
2017-12-04 15:30:08,697 [ERROR xfinity_usage.py:347 - root.error_screenshot() ] Page source saved to: /home/jantman/scratch/webdriver_fail.html
2017-12-04 15:30:08,697 [WARNING xfinity_usage.py:206 - root.do_login() ] Unable to find Remember Me button!
Traceback (most recent call last):
  File "/home/jantman/GIT/xfinity-usage/xfinity_usage.py", line 202, in do_login
    rem_me.click()
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 60, in click
    self._execute(Command.CLICK_ELEMENT)
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 370, in _execute
    return self._parent.execute(command, params)
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 173, in execute
    self.error_handler.check_response(response)
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 164, in check_response
    raise exception_class(message, screen, stacktrace)
WebDriverException: Message: u'unknown error: Element <input type="checkbox" id="remember_me" name="rm" value="1"> is not clickable at point (672, 306). Other element would receive the click: <span id="remember_me_checkbox" class="checkbox"></span>\n  (Session info: headless chrome=62.0.3202.62)\n  (Driver info: chromedriver=2.33.506092 (733a02544d189eeb751fe0d7ddca79a0ee28cce4),platform=Linux 4.13.7-1-ARCH x86_64)'
2017-12-04 15:30:08,824 [DEBUG xfinity_usage.py:221 - root.do_login() ] Clicking Sign In button
2017-12-04 15:30:10,546 [INFO xfinity_usage.py:351 - root.get() ] GET https://customer.xfinity.com/#/devices
2017-12-04 15:30:31,226 [INFO xfinity_usage.py:262 - root.get_usage_page() ] Sleeping 5s...
2017-12-04 15:30:36,267 [INFO xfinity_usage.py:267 - root.get_usage_page() ] <span class="polaris-greeting"> not in page source;login may have failed.
2017-12-04 15:30:36,267 [INFO xfinity_usage.py:262 - root.get_usage_page() ] Sleeping 5s...
2017-12-04 15:30:41,329 [INFO xfinity_usage.py:267 - root.get_usage_page() ] <span class="polaris-greeting"> not in page source;login may have failed.
2017-12-04 15:30:41,375 [CRITICAL xfinity_usage.py:293 - root.get_usage() ] Unable to find monthly usage divs on page
Traceback (most recent call last):
  File "/home/jantman/GIT/xfinity-usage/xfinity_usage.py", line 288, in get_usage
    '//*[@ng-if="device.usage"]'
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 230, in find_element_by_xpath
    return self.find_element(by=By.XPATH, value=xpath)
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 662, in find_element
    {'using': by, 'value': value})['value']
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 173, in execute
    self.error_handler.check_response(response)
  File "/home/jantman/venvs/foo/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 164, in check_response
    raise exception_class(message, screen, stacktrace)
NoSuchElementException: Message: u'no such element: Unable to locate element: {"method":"xpath","selector":"//*[@ng-if="device.usage"]"}\n  (Session info: headless chrome=62.0.3202.62)\n  (Driver info: chromedriver=2.33.506092 (733a02544d189eeb751fe0d7ddca79a0ee28cce4),platform=Linux 4.13.7-1-ARCH x86_64)'
2017-12-04 15:30:41,517 [ERROR xfinity_usage.py:339 - root.error_screenshot() ] Screenshot saved to: /home/jantman/scratch/webdriver_fail.png
2017-12-04 15:30:41,521 [ERROR xfinity_usage.py:340 - root.error_screenshot() ] Page title: XFINITY | My Account | EcoBill® Online Bill Pay
2017-12-04 15:30:41,547 [ERROR xfinity_usage.py:347 - root.error_screenshot() ] Page source saved to: /home/jantman/scratch/webdriver_fail.html
Traceback (most recent call last):
  File "/home/jantman/GIT/privatepuppet/files/xfinity2graphite.py", line 145, in <module>
    res = u.run()
  File "/home/jantman/GIT/xfinity-usage/xfinity_usage.py", line 171, in run
    res = self.get_usage()
  File "/home/jantman/GIT/xfinity-usage/xfinity_usage.py", line 295, in get_usage
    raise RuntimeError('Unable to find monthly usage divs.')
RuntimeError: Unable to find monthly usage divs.
```

Can anyone confirm this? @ericzinnikas it seems that this will break your #6 from only 4 days ago...

Just looking to see if other people are seeing this as well before I push a fix, as it will effectively revert #6.

**Comment by jantman** [comment]
Ok, I just realized that now there's a table of monthly usage:

![usage2](https://user-images.githubusercontent.com/273454/33583176-f7be05ce-d926-11e7-84f1-ff71bb679ab6.png)


The xpath would be a bit more complicated, but I assume that we can just extract the current usage from the last row of the table...

**Comment by ericzinnikas** [comment]
Yep, can confirm I no longer see the "Device Usage" area.  An xpath like `'//div[@usage-data="usage.details.usageData"]/div[2]/table/tbody/tr[last()]/td[1]'` should work, though I wasn't able to get it exactly (Comcast giving me captcha upon login)... will give it another go tomorrow.

**Comment by jantman** [comment]
@ericzinnikas I'm working on it in a branch, though I'm having some weird issues that I think are related to lazy-loading of the graph.

**Comment by jantman** [comment]
Ok, I've pushed a fix for this to master as 1.2.0
